### PR TITLE
Adds a location field for newly created users and chats; average location of users within chats calculates the chat's location

### PR DESCRIPTION
**Pull request by nlumiere**
I set a rough starting-point of within .5 degrees of latitude and longitude to be added to an existing chat.
Includes two tests for location-based chat creation.

### CODE REVIEW

File: proj/src/main/webapp/chat.js [hunk 1]
```diff
@@ -812,5 +895,10 @@ function addTag(tag, uid) {
   document.querySelector('.tag-container').insertBefore(tagContainer, tagInput)
 }
 
+
+
+
+exports.createOrJoinChat = createOrJoinChat;
```
Comment (nlumiere) [inline proj/src/main/webapp/chat.js:901]: Only reason there are so many empty lines is to reduce future merge conflicts:
TODO: (low priority) reduce number of empty lines

**Review by arringtonh — COMMENTED** [review]
lgtm

File: proj/src/main/webapp/chat.js [hunk 2]
```diff
@@ -71,24 +79,43 @@ function createNewChatWithUser(tag){
...
     return postKey;
 }
 
 // Loops through open chat rooms, adds the user to the first open chat and returns the key
-function findChatAndAddUser(snapshot, tag){
+function findChatAndAddUser(snapshot, tag, lat, long){
     var key;
+    var chatLat;
+    var chatLong;
+    const precision = .5;
```
Comment (aokeke) [inline proj/src/main/webapp/chat.js:92]: can this precision be moved to a global constant maybe in the location.js file.

**Review by aokeke — APPROVED** [review]
LGTM! This will almost certainly create merge conflicts for my PR to make keyIdDict not be global. I'll wait for this to go in and then resolve conflicts

**Review by leeannachen — APPROVED** [review]
lgtm

File: proj/src/main/webapp/chat.js (hunk 2 above)
Comment (nlumiere) [reply proj/src/main/webapp/chat.js:92]: Yes I'll do that in my next set of features

**Pull request merged**